### PR TITLE
Please do a code review

### PR DESCRIPTION
**Pull request by russtuck**

### CODE REVIEW

File: weather/www/css/style.css [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+.item {
+  background-color: #707070 !important;
+  opacity: .7;
+  /* This sets the background color*/
+}
+
+.mid {
+  text-align: center;
+}
+@media (min-height: 0) {
```
Comment (eanyanwu) [inline weather/www/css/style.css]: I don't think you need to make this a media query. The properties you set inside this media query will always be executed because a device always has a hight of at least zero.

File: weather/www/js/app.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    }
+    console.log(dayOfTheWeek(6));
+
+
+    $scope.currentDate = fullDate;
+
+
+    $scope.weather = response.data;
+    console.log($scope.weather);
+    $scope.icon = $scope.weather.currently.icon;
```
Comment (eanyanwu) [inline weather/www/js/app.js:103]: `$scope.weather = response.data` makes everything that is in the @scope.weather object available to your view (index.html). So you don't need to define scope.icon/scope.currentTemp/scope.humidity etc... in your javascript. Once you have:
`$scope.weather = response.data` 
in your js, you can then use 
`{{ weather.currently.icon }}`
in your index.html to get the icon.

File: weather/www/js/app.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    $scope.currentTemp = Math.round($scope.weather.currently.temperature);
+    $scope.precipProbability = Math.round($scope.weather.currently.precipProbability);
+    $scope.humidity = Math.round($scope.weather.currently.humidity);
+    $scope.windSpeed = Math.round($scope.weather.currently.windSpeed);
+    $scope.summary = String($scope.weather.currently.summary);
+
+    $scope.hour1 = time(1) + ", " + Math.round($scope.weather.hourly.data[1].temperature) + "°F and " +
```
Comment (eanyanwu) [inline weather/www/js/app.js:110]: Forecast.io gives you access to the timestamps at which the weather data was collected on.
So if I recall correctly, you should be able to do `$scope.weather.hourly.data[1].time` and get the time.
The time that forecast gives you is in _seconds from January 1, 1970_ (UTC). To convert it to actually useful dates, see the Javascript _Date_ documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
Note: Forecast.io gives you the time in seconds, but Javascript expects milliseconds, don't forget to convert!
**tldr;** You don't need the time(x) function and you don't need lines 77-115

File: weather/www/index.html [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+      <ion-content class="background_color" ng-controller="HomeCtrl" scroll="true">
+        <ion-slide-box show-pager="false">
+          <ion-slide>
+            <div class="row">
+              <div class="col-100">
+                <h2 sytle="font-size = 200%">
+                  {{currentDate}}
+                </h2>
+                <h2>
+                  {{summary}}
```
Comment (eanyanwu) [inline weather/www/index.html]: _Read the comment on your javascript first_
As an example, once you set `$scope.weather = response.data` in your javascript.
to access the summary here you would do `{{ weather.currently.summary }}` because summary is a property of the Json object.
Let me know if that doesn't make sense :)

**Comment by eanyanwu** [comment]
Great work Matt!
I can't imagine how tough it was working on this project not know much about css, but you did it. You even ended up with more css than I did :+1:  
From what you showed me, it looks pretty nice. Keep it up.

File: weather/www/css/style.css (hunk 1 above)
Comment (mFelgate) [inline weather/www/css/style.css]: yeah, that was me being a little lazy with the copy and paste haha

File: weather/www/js/app.js (hunk 2 above)
Comment (jordanhunt20) [inline weather/www/js/app.js:103]: Is there an advantage to that method? I wrote my code in the same way as Matt, because I just thought it made more sense to do the coding in the JS and then just use the variables in the HTML. It might also be easier to do things like rounding the returned variables in the JS, and even if you don't do that I find this method to be cleaner/easier to read, but that's probably a subjective opinion.

File: weather/www/js/app.js (hunk 3 above)
Comment (jordanhunt20) [inline weather/www/js/app.js:110]: +1 for very helpful comment. I will be returning to this when I revise my code.

File: weather/www/index.html [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+              </h2>
+            </ion-footer-bar>-->
+
+    </ion-content>
+
+  </div>
+
+</body>
+
+</html>
```
Comment (jordanhunt20) [inline weather/www/index.html:224]: Good HTML work. Easy to read, good spacing/indentation, etc.

File: weather/www/js/app.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+      "April", "May", "June", "July",
+      "August", "September", "October",
+      "November", "December"
+    ];
+
+    var days = [
+      'Sunday', 'Monday', 'Tuesday', 'Wednesday',
+      'Thursday', 'Friday', 'Saturday'
+    ];
+
```
Comment (jordanhunt20) [inline weather/www/js/app.js:42]: This is exactly what I did for days of the week. Kind of a pain, I would think it would be a default part of JS.

File: weather/www/js/app.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      $scope.weather.daily.data[3].summary;
+    $scope.day4 = dayOfTheWeek(4) + ", " + Math.round($scope.weather.daily.data[4].temperatureMax) + "°F and " +
+      $scope.weather.daily.data[4].summary;
+    $scope.day5 = dayOfTheWeek(5) + ", " + Math.round($scope.weather.daily.data[5].temperatureMax) + "°F and " +
+      $scope.weather.daily.data[5].summary;
+    $scope.day6 = dayOfTheWeek(6) + ", " + Math.round($scope.weather.daily.data[6].temperatureMax) + "°F and " +
+      $scope.weather.daily.data[6].summary;
+    $scope.day7 = dayOfTheWeek(7) + ", " + Math.round($scope.weather.daily.data[7].temperatureMax) + "°F and " +
+      $scope.weather.daily.data[7].summary;
+
```
Comment (jordanhunt20) [inline weather/www/js/app.js:149]: I think you should be able to do all of this stuff in a loop. You could do something along the lines of putting the variables in an array, and then assigning them values via looping thru the array. Something like:
var array = [$scope.day1, $scope.day2, ... ]
for (i = 1; i < 8; i++) {
    array[i] = dayOfTheWeek(i) + ", " + Math.round($scope.weather.daily.data[i].temperatureMax) + "°F and " +                          $scope.weather.daily.data[i].summary;
}

Or it might have to be something like this: 

var array = [day1, day2, ... ]
for (i = 1; i < 8; i++) {
    $scope.array[i] = dayOfTheWeek(i) + ", " + Math.round($scope.weather.daily.data[i].temperatureMax) + "°F and " +
    $scope.weather.daily.data[i].summary;
}

I'm definitely not sure exactly how to do this, but something along those lines would work, I think. This should be possible for a lot of this file.

**Comment by jordanhunt20** [comment]
Good job with the fundamentals of the code, e.g. variable naming, consistent camelCase, indentation, etc. The front page looks great in my opinion as well.

I would recommend (as I commented) using loops to assign variables in a few places; being lazy is good in programming, and this should allow for easier changes to the code since a change in one place will affect all of the variables in the loop.

I honestly really like your app and am somewhat inspired by it to work on the css for my own app. There were not really any major negatives that stood out to me.

File: weather/www/js/app.js (hunk 3 above)
Comment (mFelgate) [inline weather/www/js/app.js:110]: That is very useful indeed haha

File: weather/www/index.html (hunk 4 above)
Comment (mFelgate) [inline weather/www/index.html]: It seems its basically the same thing, except there more code in the html then in the java, I'm happy to change if your way is better practice though.

**Comment by mFelgate** [comment]
thanks Jordan!

File: weather/www/js/app.js (hunk 7 above)
Comment (mFelgate) [inline weather/www/js/app.js:149]: So i tried using a for loop, however it says the variable becomes undefined for both for loops, I will look for a solution at a later date
